src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -606,41 +648,53 @@ function proxyFirebaseApp(
       firestoreTarget.settings(emulatorSettings);
 
       new EmulatorLog("DEBUG", "set-firestore-settings", "", emulatorSettings).log();
-    } else if (!frb.ports.firestore && frb.triggerId) {
-      new EmulatorLog(
-        "WARN",
-        "runtime-status",
-        "The Cloud Firestore emulator is not running so database operations will fail with a " +
-          "'default credentials' error."
-      ).log();
     }
 
     hasInitializedFirestore = true;
   };
 
-  const appProxy = new Proxied<admin.app.App>(original);
-  appProxy.when("firestore", (target: any) => {
-    // TODO: this 'typeof admin.firestore' should probably be 'Firestore' from @google-cloud/firestore
-    //       but I can't get all the type checking to play nice.
-    const firestoreProxy = new Proxied<typeof admin.firestore>(target.firestore);
-    return firestoreProxy
-      .applied(() => {
-        return new Proxied(target.firestore())
-          .when("settings", (firestoreTarget) => {
-            return (settings: any) => {
-              initializeFirestoreSettings(firestoreTarget, settings);
-            };
-          })
-          .any((firestoreTarget, field) => {
-            initializeFirestoreSettings(firestoreTarget, {});
-            return firestoreProxy.getOriginal(firestoreTarget, field);
-          })
-          .finalize();
-      })
-      .finalize();
-  });
+  const firestoreProxy = new Proxied<typeof admin.firestore>(target.firestore);
+  return firestoreProxy
+    .applied(() => {
+      return new Proxied(target.firestore())
+        .when("settings", (firestoreTarget) => {
+          return (settings: any) => {
+            initializeFirestoreSettings(firestoreTarget, settings);
+          };
+        })
+        .any((firestoreTarget, field) => {
```

Comment:
Would `admin.firestore.FieldValue` also trigger this? Is it working as intended?